server/room-battle.js

```diff
@@ -783,7 +783,15 @@ class RoomBattle extends RoomGames.RoomGame {
 			parentGame.onBattleWin(this.room, winnerid);
 		}
 		// If the room's replay was hidden, disable users from joining after the game is over
-		if (this.room.hideReplay) this.room.modjoin = '%';
+		if (this.room.hideReplay) {
+			this.room.modjoin = '%';
+			this.room.isPrivate = 'hidden';
```

Comment:
I'm going off the assumption that tracking `privateSetter` doesn't really matter here because it's not the end of the world if someone does `/publicroom` after - the main concern is the replay not being password protected.

Reply:
This shouldn't affect whether or not the replay is password protected – `hideReplay` should be enough.

Reply:
> NOTE: this change does not fix private battles not properly getting passworded, but closes a few other loopholes that were found while investigating.

Seriously, its like you don't read anything I write...

Reply:
I suppose I understand which loophole this is.

Reply:
And yes, at this point I've been bombarded with too much information to closely read everything said to me. I might misread things.

Reply:
I was originally opposed, but I guess it makes sense for the room to be hidden.